string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,38 @@
+function titleCase(title) {
+  const exceptions = ['a', 'an', 'the', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to', 'and'];
+  const updatedTitle = [];
+  let str;
+  let specialCharPosition;
+  let specialChar;
+  if (title.includes(':')) {
+    specialCharPosition = title.indexOf(':');
+    specialChar = ':';
+    str = title.toLowerCase().split(':');
+  } else if (title.includes('-')) {
+    str = title.toLowerCase().split('-');
+  }
+  str = title.toLowerCase().split(' ');
+
+  for (let i = 0; i < str.length; i++) {
+    updatedTitle[i] = str[i][0].toUpperCase() + str[i].slice(1).toLowerCase();
+
```

Comment:
there needs to be a conditional check inside this `for` loop to check for hyphenatd words, like `in-depth`. In this case, both words before and after the `-` need to be capitalized.

You can do something like this: 
```
    if (str[i].includes('-')) {
      const hyphenSplit = str[i].split('-');
      const hyphenCapitalized  = hyphenSplit
        .map(word => word[0].toUpperCase() + word.slice(1).toLowerCase())
        .join('-');
      updatedTitle[i] = hyphenCapitalized;
    }
```

this checks if the word in the loop contains a `-`, if it does it will split that word into an array of the words with out the `-` (ie. `['in', 'depth']`.
Then we use the `map` method to iterate through the newly created array and capitalize each word separately, and then `join` them back into a single string, replacing the `-` between the words.
Lastly, we assign the properly capitalized string back to `updateTitle[i]` to work as you had it with the rest of the `for` loop.

Hope this explanation helps!